go-controller/pkg/informer/informer.go

```diff
@@ -61,6 +61,14 @@ func DiscardAllUpdates(old, new interface{}) bool {
 	return false
 }
 
+// EventHandlerFactory is function that creates new event handlers
+type EventHandlerFactory func(
```

Comment:
this isn't really a factory. It's just a type abstraction so you can now pass either of your new event handlers. Let's call it something else.

Reply:
I'm going to call it `Tim` 😂 
Just kidding, I'll push an update with a more sensible name.